src/features/users/usersSlice.ts

```diff
@@ -0,0 +1,23 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { User } from '../../types/User';
+
+type UsersState = {
+  users: User[];
+};
+
+const initialState: UsersState = {
+  users: [],
+};
+
+const usersSlice = createSlice({
+  name: 'users',
+  initialState,
+  reducers: {
+    set: (state, action: PayloadAction<User[]>) => {
+      state.users.push(...action.payload);
```

Comment:
Instead of using 'push' to add elements to 'state.users', you should directly assign 'action.payload' to 'state.users' for better readability and to avoid potential issues.